File: android/src/main/java/dev/nittenapps/capacitor/encode/EncodeEnginePlugin.java
package dev.nittenapps.capacitor.encode;

import androidx.annotation.NonNull;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "EncodeEngine")
public class EncodeEnginePlugin extends Plugin {
    @PluginMethod
    public void checkDigits(@NonNull PluginCall call) {
        String input = call.getString("input");

        JSObject ret = new JSObject();
        ret.put("value", EncodeEngine.checkDigits(input));
        call.resolve(ret);
    }

    @PluginMethod
    public void decode(@NonNull PluginCall call) {
        String key = call.getString("key");
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", EncodeEngine.decode(key, value));
        call.resolve(ret);
    }

    @PluginMethod
    public void encode(@NonNull PluginCall call) {
        String key = call.getString("key");
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", EncodeEngine.encode(key, value));
        call.resolve(ret);
    }

    @PluginMethod
    public void sign(@NonNull PluginCall call) {
        String value = call.getString("value");
        assert value != null;

        JSObject ret = new JSObject();
        ret.put("value", EncodeEngine.sign(bridge.getContext(), value));
        call.resolve(ret);
    }
}

File: ios/Sources/EncodeEnginePlugin/EncodeEnginePlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(EncodeEnginePlugin)
public class EncodeEnginePlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "EncodeEnginePlugin"
    public let jsName = "EncodeEngine"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "decode", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "encode", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = EncodeEngine()

    @objc func checkDigits(_ call: CAPPluginCall) {
        let input = call.getString("input") ?? ""

        call.resolve([
            "value": implementation.checkDigits(input)
        ])
    }

    @objc func decode(_ call: CAPPluginCall) {
        let key = call.getString("key") ?? ""
        let value = call.getString("value") ?? ""

        call.resolve([
            "value": implementation.decode(key, value)
        ])
    }

    @objc func encode(_ call: CAPPluginCall) {
        let key = call.getString("key") ?? ""
        let value = call.getString("value") ?? ""

        call.resolve([
            "value": implementation.encode(key, value)
        ])
    }
}
